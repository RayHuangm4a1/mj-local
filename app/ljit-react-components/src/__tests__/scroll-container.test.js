import React from 'react';
import { shallow, mount, } from 'enzyme';
import { MemoryRouter as Router, } from 'react-router-dom';
import ScrollContainer from '../components/scroll-container';

const mockList = `
	<ul>
		<li key='1'>mock item 1</li>
		<li key='2'>mock item 2</li>
		<li key='3'>mock item 3</li>
	</ul>`;

describe('ScrollContainer', () => {
	it('should handle default props', () => {
		const {
			buttonText,
			isButtonDisabled,
			onClick,
		} = ScrollContainer.defaultProps;

		expect(buttonText).toEqual('更多');
		expect(isButtonDisabled).toEqual(false);
		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const buttonText = 'mock-text';
		const className = 'mock-class';
		const style = { maxWidth: '200px', };
		const isButtonDisabled = false;
		const onClick = () => console.log('click');

		const wrapper = shallow(
			<ScrollContainer
				buttonText={buttonText}
				className={className}
				style={style}
				isButtonDisabled={isButtonDisabled}
				onClick={onClick}
			>
				{mockList}
			</ScrollContainer>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-scroll-container', () => {
		const wrapper = shallow(
			<ScrollContainer>
				{mockList}
			</ScrollContainer>
		);

		expect(wrapper.hasClass('ljit-scroll-container')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(
			<ScrollContainer className={className}>
				{mockList}
			</ScrollContainer>
		);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const children = mockList;
		const buttonText = 'mock-text';
		const className = 'mock-class';
		const style = { maxWidth: '200px', };
		const isButtonDisabled = false;
		const onClick = () => console.log('click');

		const wrapper = mount(
			<ScrollContainer
				buttonText={buttonText}
				className={className}
				style={style}
				isButtonDisabled={isButtonDisabled}
				onClick={onClick}
			>
				{children}
			</ScrollContainer>
		);

		expect(wrapper.props().buttonText).toBe(buttonText);
		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().style).toMatchObject(style);
		expect(wrapper.props().isButtonDisabled).toEqual(isButtonDisabled);
		expect(wrapper.props().children).toBe(children);
		expect(wrapper.props().onClick).toEqual(onClick);
	});

	it('should handle onClick event', () => {
		const onClick = jest.fn();
		const wrapper = mount(
			<Router>
				<ScrollContainer onClick={onClick}>
					{mockList}
				</ScrollContainer>
			</Router>
		);
		const button = wrapper.find('Button').at(0);

		button.simulate('click');

		expect(onClick).toHaveBeenCalledTimes(1);
		expect(onClick).toHaveBeenCalled();
	});
});
