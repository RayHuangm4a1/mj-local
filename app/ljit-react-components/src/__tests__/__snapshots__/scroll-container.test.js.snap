// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollContainer should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollContainer
    buttonText="mock-text"
    className="mock-class"
    isButtonDisabled={false}
    onClick={[Function]}
    style={
      Object {
        "maxWidth": "200px",
      }
    }
  >
    
	&lt;ul&gt;
		&lt;li key='1'&gt;mock item 1&lt;/li&gt;
		&lt;li key='2'&gt;mock item 2&lt;/li&gt;
		&lt;li key='3'&gt;mock item 3&lt;/li&gt;
	&lt;/ul&gt;
  </ScrollContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-scroll-container__body"
        >
          
	&lt;ul&gt;
		&lt;li key='1'&gt;mock item 1&lt;/li&gt;
		&lt;li key='2'&gt;mock item 2&lt;/li&gt;
		&lt;li key='3'&gt;mock item 3&lt;/li&gt;
	&lt;/ul&gt;
        </div>,
        <Button
          className="ljit-scroll-container__button"
          color=""
          disabled={false}
          icon="down"
          isFullWidth={true}
          onClick={[Function]}
          outline="text"
        >
          mock-text
        </Button>,
      ],
      "className": "ljit-scroll-container mock-class",
      "style": Object {
        "maxWidth": "200px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "
	<ul>
		<li key='1'>mock item 1</li>
		<li key='2'>mock item 2</li>
		<li key='3'>mock item 3</li>
	</ul>",
          "className": "ljit-scroll-container__body",
        },
        "ref": null,
        "rendered": "
	<ul>
		<li key='1'>mock item 1</li>
		<li key='2'>mock item 2</li>
		<li key='3'>mock item 3</li>
	</ul>",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "mock-text",
          "className": "ljit-scroll-container__button",
          "color": "",
          "disabled": false,
          "icon": "down",
          "isFullWidth": true,
          "onClick": [Function],
          "outline": "text",
        },
        "ref": null,
        "rendered": "mock-text",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-scroll-container__body"
          >
            
	&lt;ul&gt;
		&lt;li key='1'&gt;mock item 1&lt;/li&gt;
		&lt;li key='2'&gt;mock item 2&lt;/li&gt;
		&lt;li key='3'&gt;mock item 3&lt;/li&gt;
	&lt;/ul&gt;
          </div>,
          <Button
            className="ljit-scroll-container__button"
            color=""
            disabled={false}
            icon="down"
            isFullWidth={true}
            onClick={[Function]}
            outline="text"
          >
            mock-text
          </Button>,
        ],
        "className": "ljit-scroll-container mock-class",
        "style": Object {
          "maxWidth": "200px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
	<ul>
		<li key='1'>mock item 1</li>
		<li key='2'>mock item 2</li>
		<li key='3'>mock item 3</li>
	</ul>",
            "className": "ljit-scroll-container__body",
          },
          "ref": null,
          "rendered": "
	<ul>
		<li key='1'>mock item 1</li>
		<li key='2'>mock item 2</li>
		<li key='3'>mock item 3</li>
	</ul>",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "mock-text",
            "className": "ljit-scroll-container__button",
            "color": "",
            "disabled": false,
            "icon": "down",
            "isFullWidth": true,
            "onClick": [Function],
            "outline": "text",
          },
          "ref": null,
          "rendered": "mock-text",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
